### PR TITLE
remove the logic of reduce gradient for XLA, since Accelerate will handle it automatically.

### PR DESCRIPTION
**Pull request by ji-huazhong**
# What does this PR do?

As the title says. 
See https://github.com/huggingface/accelerate/pull/1926 for more information.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

cc @muellerzr

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
Thanks for updating! 

LGTM - let's wait until we get a second opinion from @muellerzr before merging

File: src/transformers/trainer.py
Comment (muellerzr) [inline src/transformers/trainer.py:1875]: I feel like there's a bit more to this actually..... Because what we really need to have happen is any calls to `self.scaler.unscale_` should really now use `accelerator.unscale_gradients(self.optimizer)`.

Otherwise we won't be unscaling the gradients at all here. I know that's a bit more than the initial PR and will take a bit more work, but does this make sense? (This is the chunk of code where you put our `self.reduce` in the Accelerate PR). So let's try and do that as part of this PR if possible as well please :)

**Review by muellerzr — COMMENTED** [review]
Thanks! While this is right, I think this is only half right and we need more to do with this. See my note, it should be as simple as replacing `self.scaler.unscale_` with `self.accelerator.unscale_gradients(self.optimizer)`

File: src/transformers/trainer.py
Comment (statelesshz) [reply src/transformers/trainer.py:1875]: > I feel like there's a bit more to this actually..... Because what we really need to have happen is any calls to self.scaler.unscale_ should really now use accelerator.unscale_gradients(self.optimizer).

It sounds good to me. It seems that there are some specific details mentioned here, which might need to be discussed further. If I'm not mistaken, `accelerator.unscale_gradients` is implicitly called when calling `self.accelerator.clip_grad_norm_`

https://github.com/huggingface/transformers/blob/fb7d246951d5f60aa36a7958841dfea72f51fc6b/src/transformers/trainer.py#L1931-L1935

I think the logic of `self.scaler.unsacle_(self.optimizer)` here only applies to ShardedDDP as we can see `self.do_grad_scaling` and `self.scaler` are initialized in the following code block
https://github.com/huggingface/transformers/blob/fb7d246951d5f60aa36a7958841dfea72f51fc6b/src/transformers/trainer.py#L629-L640

In order to be compatible with ShardedDDP, the current implementation is somewhat fragmented, so removing the ShardedDDP-related code first may help clarify things. WDYT @muellerzr

Comment (muellerzr) [reply src/transformers/trainer.py:1875]: Yes, especially as Accelerate uses `ShardedGradScaler` https://github.com/huggingface/accelerate/blob/main/src/accelerate/accelerator.py#L427-L434

Comment (statelesshz) [reply src/transformers/trainer.py:1875]: I apologize for not communicating my thoughts with enough precision. In transformers, `ShardedGradScaler` is imported from `from fairscale.optim.grad_scaler import ShardedGradScaler` not `from torch.distributed.fsdp.sharded_grad_scaler import ShardedGradScaler` @muellerzr

Comment (muellerzr) [reply src/transformers/trainer.py:1875]: Aren't we removing this anyways via https://github.com/huggingface/transformers/pull/25702?

Comment (statelesshz) [reply src/transformers/trainer.py:1875]: Yes, this PR should closed

**Pull request closed without merging**
